### PR TITLE
Draft to change oteltest to use the sdk

### PR DESCRIPTION
**Pull request by bogdandrutu**
Fixes https://github.com/open-telemetry/opentelemetry-go/issues/2072

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

**Comment by bogdandrutu** [comment]
This PR is to prove how easily an oteltest package can be implemented on top of the SDK, and avoid 1000+ duplicate code lines.

File: go.mod [hunk 1]
```diff
@@ -6,6 +6,7 @@ require (
 	github.com/google/go-cmp v0.5.6
 	github.com/stretchr/testify v1.7.0
 	go.opentelemetry.io/otel/oteltest v1.0.0-RC1
+	go.opentelemetry.io/otel/sdk v1.0.0-RC1
```
Comment (Aneurysm9) [inline go.mod:9]: This is the API module and `MUST NOT` depend on any SDK.

**Review by Aneurysm9 — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (bogdandrutu) [reply go.mod:9]: The API module probably MUST NOT depend on the oteltest :D

Comment (bogdandrutu) [reply go.mod:9]: @Aneurysm9 if you resolve https://github.com/open-telemetry/opentelemetry-go/issues/2077 then this will go away :D

**Comment by MrAlias** [comment]
One of the intended purposes of the `oteltest` package is for it to be used by instrumentation authors to test their code. This means that it should not itself depend on the SDK as it will mean any use of it by the instrumentation authors would bring along that dependency. The approach taken here would violate this dependency requirement. Not only would this put a burden on our users to replace the `oteltest` package, but us as well:

- https://github.com/open-telemetry/opentelemetry-go-contrib/blob/262f81f753760cd1333d1925918300be3f01e77e/instrumentation/google.golang.org/grpc/otelgrpc/go.mod#L12
- https://github.com/open-telemetry/opentelemetry-go-contrib/blob/262f81f753760cd1333d1925918300be3f01e77e/instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo/go.mod#L12
- https://github.com/open-telemetry/opentelemetry-go-contrib/blob/262f81f753760cd1333d1925918300be3f01e77e/instrumentation/net/http/otelhttp/go.mod#L13
- https://github.com/open-telemetry/opentelemetry-go-contrib/blob/262f81f753760cd1333d1925918300be3f01e77e/instrumentation/net/http/httptrace/otelhttptrace/go.mod#L12
- https://github.com/open-telemetry/opentelemetry-go-contrib/blob/262f81f753760cd1333d1925918300be3f01e77e/instrumentation/github.com/aws/aws-sdk-go-v2/otelaws/go.mod#L17
- https://github.com/open-telemetry/opentelemetry-go-contrib/blob/262f81f753760cd1333d1925918300be3f01e77e/instrumentation/github.com/labstack/echo/otelecho/go.mod#L16
- https://github.com/open-telemetry/opentelemetry-go-contrib/blob/262f81f753760cd1333d1925918300be3f01e77e/instrumentation/github.com/go-kit/kit/otelkit/go.mod#L10
- https://github.com/open-telemetry/opentelemetry-go-contrib/blob/262f81f753760cd1333d1925918300be3f01e77e/instrumentation/github.com/bradfitz/gomemcache/memcache/otelmemcache/go.mod#L12
- https://github.com/open-telemetry/opentelemetry-go-contrib/blob/262f81f753760cd1333d1925918300be3f01e77e/instrumentation/github.com/gin-gonic/gin/otelgin/go.mod#L16
- https://github.com/open-telemetry/opentelemetry-go-contrib/blob/262f81f753760cd1333d1925918300be3f01e77e/instrumentation/github.com/gocql/gocql/otelgocql/go.mod#L14
- https://github.com/open-telemetry/opentelemetry-go-contrib/blob/262f81f753760cd1333d1925918300be3f01e77e/instrumentation/github.com/emicklei/go-restful/otelrestful/go.mod#L17
- https://github.com/open-telemetry/opentelemetry-go-contrib/blob/262f81f753760cd1333d1925918300be3f01e77e/instrumentation/github.com/Shopify/sarama/otelsarama/go.mod#L12
- https://github.com/open-telemetry/opentelemetry-go-contrib/blob/262f81f753760cd1333d1925918300be3f01e77e/instrumentation/github.com/gorilla/mux/otelmux/go.mod#L17
- https://github.com/open-telemetry/opentelemetry-go-contrib/blob/262f81f753760cd1333d1925918300be3f01e77e/instrumentation/github.com/astaxie/beego/otelbeego/go.mod#L18
- https://github.com/open-telemetry/opentelemetry-go-contrib/blob/262f81f753760cd1333d1925918300be3f01e77e/instrumentation/gopkg.in/macaron.v1/otelmacaron/go.mod#L15

That said, I think it is possible to unify and use the same implementation of a `Tracer`. It will need to be an implementation that the SDK would also take a dependency on. Meaning we could have an implementation in an internal package that both the sdk and `oteltest` would use.

**Pull request closed without merging**